### ISSUE TITLE
OpenAPI v2 converter: referenced parameter produces a validation error

### ISSUE THREAD

**Issue by jmini** [swagger-api/swagger-parser#1146]
With this OpenAPI v2 spec:
```yaml
swagger: "2.0"
info:
  description: "Test"
  version: "1.0.0"
  title: "Swagger v2 example"
  termsOfService: "http://swagger.io/terms/"
  contact:
    email: "apiteam@swagger.io"
  license:
    name: "Apache 2.0"
    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
host: "petstore.swagger.io"
basePath: "/v2"
schemes:
- "https"
- "http"
paths:
  /brands/{brand-id}/stations:
    get:
      tags:
        - Brand
      summary: Returns a list of the brand's stations
      operationId: getBrandStations
      parameters:
        - $ref: "#/parameters/brand-id"
        - in: query
          name: available-to-client-only
          required: true
          type: boolean
      responses:
        200:
          description: OK
parameters:
  brand-id:
    in: path
    description: this is a test
    name: brand-id
    required: true
    type: string
```

When I parse it like this:

```java
OpenAPIParser openApiParser = new OpenAPIParser();
ParseOptions options = new ParseOptions();
options.setResolve(true);
options.setFlatten(true);

SwaggerParseResult parseResult = openApiParser.readLocation(inputSpec, null, options);
for (String string : parseResult.getMessages()) {
    System.err.println(string);
}
```

I get this error:
```
attribute paths.'/brands/{brand-id}/stations'. Declared path parameter brand-id needs to be defined as a path parameter in path or operation level
```

This is wrong, because the parameter is defined.

Swagger-UI is interpreting the spec correctly:

<img width="598" alt="Screenshot 2019-07-08 at 20 07 06" src="https://user-images.githubusercontent.com/1222165/60832382-07e6ca80-a1bc-11e9-85af-d02376484087.png">

----

As comparison the corresponding OpenAPI v3 spec is working fine:

```yaml
openapi: 3.0.1
info:
  title: Example ref param
  version: '1.0'
servers:
  - url: 'http://localhost:9999/'
paths:
  /brands/{brand-id}/stations:
    get:
      tags:
        - Brand
      summary: Returns a list of the brand's stations
      operationId: getBrandStations
      parameters:
        - $ref: "#/components/parameters/brand-id"
        - in: query
          name: available-to-client-only
          required: true
          schema:
            type: boolean
      responses:
        200:
          description: OK
components:
  parameters:
    brand-id:
      in: path
      required: true
      name: brand-id
      schema:
        type: string
```

**Comment by JFCote** [comment]
I've done some digging and found that the warning starts to show on this line:

`SwaggerDeserializationResult result = new SwaggerParser().readWithInfo(url, convert(auths), resolve);`

This use the old swagger-parser v1.0.45 so I guess the bug is in the old parser.

I will try to find the source next week but if someone else is looking at this, that might help!

**Comment by JFCote** [comment]
This bug could be fix very quickly by applying the fix in PR #977 to the `v1` branch. Then a quick release and use the newly created version in the master should fix all these warnings.

**Comment by gracekarina** [comment]
Hi @JFCote @jmini please let us know if this issue keeps happening. thanks for the PR.
